### PR TITLE
Rewrote Nuclei Commands

### PR DESCRIPTION
**Pull request by payoub125**
## Rewrote Nuclei Tools

* Detect API Application

  - Swagger
  - K8s (Separated this tool into 2 to follow the struct of the others (Detect + Enumerate)
  - GRPC
  - GraphQL
  - FastAPI

* Detect Cloud Bucket

  - Azure Blob
  - Aws S3

### Command

```
% method app detect -h                                                                                                             
Perform a detection scan against a target using specified types.
		
The detection command identifies the type of web application running on the target URL.
It supports detecting different resource types including API applications (Swagger, gRPC, GraphQL, K8s), 
cloud buckets (AWS S3, Azure Blob), and web applications. The command accepts a list of modules to run
for the specified resource type.

Usage:
  webscan app detect [flags]

Flags:
  -h, --help                  help for detect
      --module strings        Modules to run
      --resourcetype string   Resource type to detect
      --successfulonly        Only show successful attempts
      --targets strings       URL target to perform detect against
      --timeout int           Timeout per request (seconds) (default 5)

Global Flags:
  -o, --output string        Output format (signal, json, yaml). Default value is signal (default "signal")
  -f, --output-file string   Path to output file. If blank, will output to STDOUT
  -q, --quiet                Suppress output
  -v, --verbose              Verbose output
```

### Data Example

```
{
    "content": {
        "resourseType": "CLOUDBUCKET",
        "resourses": [
            {
                "target": "https://leaky.bucket.method.directory",
                "attempts": [
                    {
                        "name": {
                            "type": "CloudBucketModule",
                            "value": "AWS_S3"
                        },
                        "AttemptInfo": [
                            {
                                "request": {
                                    "baseUrl": "https://leaky.bucket.method.directory",
                                    "path": "/",
                                    "method": "GET"
                                },
                                "response": {
                                    "statusCode": 200,
                                    "responseHeaders": {
                                        "Content-Type": "application/xml",
                                        "Date": "Tue, 26 Nov 2024 14:41:00 GMT",
                                        "Server": "AmazonS3",
                                        "X-Amz-Bucket-Region": "us-east-1",
                                        "X-Amz-Id-2": "oBq+.../06Kuk9H79+YmaljgCh0YPjBK2C8=",
                                        "X-Amz-Request-Id": "...."
                                    },
                                    "responseBody": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<ListBucketResu...ucketResult>"
                                }
                            }
                        ],
                        "finding": true,
                        "timestamp": "2024-11-26T09:40:59-05:00"
                    }
                ]
            }
        ],
        "config": {
            "targets": [
                "https://leaky.bucket.method.directory"
            ],
            "modules": [
                {
                    "type": "CloudBucketModule",
                    "value": "AWS_S3"
                }
            ],
            "resourceType": "CLOUDBUCKET",
            "timeout": 5,
            "successfulOnly": false
        }
    },
    "started_at": "0001-01-01T00:00:00Z",
    "completed_at": "2024-11-26T09:40:59.193933-05:00",
    "status": 0
}
```

### CODE REVIEW

**Comment by payoub125** [comment]
The files I changed arent even appearing. Ill split this into 2 PRs. One that deletes Nuclie and one that adds back all the new code

File: cmd/app.go [hunk 1]
```diff
@@ -13,13 +17,87 @@ func (a *WebScan) InitAppCommand() {
...
+			report, err := engine.Launch(cmd.Context())
+			if err != nil {
+				a.OutputSignal.AddError(err)
+			}
+			a.OutputSignal.Content = report
+		},
+	}
+
+	detectCmd.Flags().StringSlice("targets", []string{}, "URL target to perform detect against")
+	detectCmd.Flags().String("resourcetype", "", "Resource type to detect")
```
Comment (sean-hacker) [inline cmd/app.go]: How will a user know what valid resource types are? I think we need to update this string. (Ack that it's in the top level description, but I think having it here will make it a lot easier for the user. Maybe we see if we can sprintf it and dynamically generate the list of valid values)

File: cmd/app.go [hunk 2]
```diff
@@ -13,13 +17,87 @@ func (a *WebScan) InitAppCommand() {
...
+	detectCmd.Flags().StringSlice("modules", []string{}, "Modules to run")
```
Comment (sean-hacker) [inline cmd/app.go]: Same as above. What are valid modules? Also, it's not clear what the difference between a resource type and a module is or how they interact.

File: cmd/app.go [hunk 3]
```diff
@@ -160,3 +238,53 @@ HTTP methods, query parameters, and authentication mechanisms.`,
...
+	if err != nil {
+		return nil, err
+	}
+	return &resourceTypeEnum, nil
+}
+
+func validateDetectResourseModuleSelection(resourceType webscan.DetectResourceType, modules []string) ([]*webscan.DetectResourceModule, error) {
+	moduleEnums := []*webscan.DetectResourceModule{}
+	if len(modules) == 0 {
+		return nil, nil
```
Comment (sean-hacker) [inline cmd/app.go:271]: Should we expect/require at least one module?

Comment (payoub125) [reply cmd/app.go:271]: No - by default all modules are run for whichever resourse type you set

File: cmd/app.go (hunk 1 above)
Comment (payoub125) [reply cmd/app.go]: A lot cleaner 

```
% method app detect -h                                                                                                 
Perform a detection scan against a target using specified types.
		
The detection command identifies the type of web application running on the target URL.
It supports detecting different resource types including API applications (FastAPI, Swagger, gRPC, GraphQL, K8s), and 
cloud buckets (AWSS3, AzureBlob). The command accepts a list of modules to run
for the specified resource type.

Usage:
  webscan app detect [flags]

Flags:
  -h, --help                  help for detect
      --modules strings       Modules to run (APIApplication: FASTAPI, GRAPHQL, GRPC, SWAGGER, K8S; CloudBucket: AWSS3, AZUREBLOB)
      --resourcetype string   Resource type to detect (APIAPPLICATION, CLOUDBUCKET)
      --successfulonly        Only show successful attempts
      --targets strings       URL target to perform detect against
      --timeout int           Timeout per request (seconds) (default 5)

Global Flags:
  -o, --output string        Output format (signal, json, yaml). Default value is signal (default "signal")
  -f, --output-file string   Path to output file. If blank, will output to STDOUT
  -q, --quiet                Suppress output
  -v, --verbose              Verbose output
```

File: cmd/app.go [hunk 4]
```diff
@@ -13,13 +18,104 @@ func (a *WebScan) InitAppCommand() {
...
 	}
 
 	a.RootCmd.AddCommand(a.AppCmd)
+	a.initDetectCommand()
 	a.initEnumerateCommand()
 }
 
+func (a *WebScan) initDetectCommand() {
+	detectCmd := &cobra.Command{
+		Use:   "detect",
```
Comment (methodsecuritysj) [inline cmd/app.go]: can we change this to fingerprint

I think that is a more accurate word

Comment (payoub125) [reply cmd/app.go]: Sounds good - We should probably update the naming for `webscan fingerprint` in that case which right now does the below. 

```
// InitFingerprintCommand initializes the fingerprint command for the webscan CLI. This command is used to perform a fingerprint
// against a URL target, capturing data TLS and HTTP methods that exist on the URL.
```

I think nesting it under url would make sense -> `webscan url fingerprint` lmk what you think

Comment (payoub125) [reply cmd/app.go]: This way we can rename the folders + files to be `appfingerprint` and `urlfingerprint` so we keep the naming 'orginization' of the repo

Comment (payoub125) [reply cmd/app.go]: I may be over complicating this, ill just rename all the folders + files to appfingerprint

**Review by methodsecuritysj — APPROVED** [review]

**Pull request merged**